### PR TITLE
chore(e2e): added e2e for create proposal

### PR DESCRIPTION
**Pull request by veralygit**
#### What this PR does / why we need it:
As titled 
Added tests : 

- check that breadcrumb is visible
- check that header title is visible
- check that Getting Started info is visible 
- check that create proposal step 1 header and description is visible 
- check radio group checks and unchecks for cfp
- check radio group checks and unchecks for dfip
- check that `Name of Proposal` input is visible for DFIP proposal creation and error msg are available
- check that `Discussion` input is visible for DFIP proposal creation and error msg are available
- check that `Amount Requested, Cycles and Receiving address`fields are not present in DFIP proposal creations
- check that `Name of Proposal` input is visible for CFP proposal creation and error msg are available
- check that `Discussion` input is visible for CFP proposal creation and error msg are available
- check that `Amount Requested` input is visible for CFP proposal creation and error msg are available
- check that `Cycles` input is visible for CFP proposal creation and error msg are available
- check that `Receiving address` input is visible for CFP proposal creation and error msg are available
- check input clear using clear button
- check filling up of input fields for CFP proposal and click on Review Proposal button 
- check that create proposal step 2 header and description is visible 
- check that `Name of Proposal` details is visible for CFP proposal creation and error msg are available
- check that `Discussion` details is visible for CFP proposal creation and error msg are available
- check that `Amount Requested` details is visible for CFP proposal creation and error msg are available
- check that `Cycles` details is visible for CFP proposal creation and error msg are available
- check that `Receiving address` details is visible for CFP proposal creation and error msg are available
- check that step 2 confirm details button is visible and clickable
- check that create proposal step 3 header and description is visible 
- check that step 3 command generated is visible and matches expected command
- check that step 3 command can be copied to clipboard
- check that step 3 info note is visible
- check editing of proposals
- check cancelling of editing of proposals
- check switching of proposal type when creating proposal
- check that we are able to create DFIP proposal

#### Which issue(s) does this PR fixes?:

Fixes #

#### Sample Links & Screenshots:

Link:

<details>
<summary>Desktop Screenshot</summary>

</details>
<details>
<summary>Mobile Screenshot</summary>

</details>

#### Developer Checklist:

- [ ] Read your code changes at least once
- [ ] Tested on multiple web browsers
- [ ] Tested responsiveness (e.g, iPhone, iPad, Desktop)
- [ ] No console errors
- [ ] Unit tests\*
- [x] Added e2e tests\*

### CODE REVIEW

File: cypress/e2e/pages/governance/create/index.spec.ts [hunk 1]
```diff
@@ -0,0 +1,532 @@
...
+          .should("be.visible")
+          .within(() => {
+            cy.findByTestId(
+              "Governance.Create.Step1.RadioGroup.CFP.Unchecked"
+            ).should("be.visible");
+          });
+      });
+    });
+
+    it("should have step 1 textarea inputs for cfp and dfip", () => {
```
Comment (kyleleow) [inline cypress/e2e/pages/governance/create/index.spec.ts:113]: let's split into individual `it()` clauses for each input

File: cypress/e2e/pages/governance/create/index.spec.ts [hunk 2]
```diff
@@ -0,0 +1,532 @@
...
+      });
+      cy.findByTestId("Governance.Create.Step1.ClearForm").click();
+      cy.findByTestId(
+        "Governance.Create.Step1.TextArea.ReceivingAddress"
+      ).within(() => {
+        cy.get("textarea").should("have.text", "");
+      });
+    });
+
+    it("should have step 2", () => {
```
Comment (kyleleow) [inline cypress/e2e/pages/governance/create/index.spec.ts:262]: split into individual `it()`, this is too broad, its quite hard to review

File: cypress/e2e/pages/governance/create/index.spec.ts [hunk 3]
```diff
@@ -0,0 +1,532 @@
...
+            "A command line will be generated once all details are confirmed"
+          );
+
+        cy.findByTestId(
+          "Governance.Create.Step2.ReviewProposal.ConfirmDetails"
+        ).click();
+      });
+    });
+
+    it("should have step 3", () => {
```
Comment (kyleleow) [inline cypress/e2e/pages/governance/create/index.spec.ts:357]: same thing

File: cypress/e2e/pages/governance/create/index.spec.ts (hunk 1 above)
Comment (kyleleow) [reply cypress/e2e/pages/governance/create/index.spec.ts:113]: Nne scenario for you to consider: if in the future, only `amount requested` field is not found, and it break this test case, you wont know what is missing just from this title `should have step 1 textarea inputs for cfp and dfip`

Comment (kyleleow) [reply cypress/e2e/pages/governance/create/index.spec.ts:113]: spending extra time now to write a finer title will save you or other future maintainer time

File: cypress/e2e/pages/governance/create/index.spec.ts [hunk 4]
```diff
@@ -0,0 +1,593 @@
...
+        .should("be.visible")
+        .within(() => {
+          cy.findByTestId(
+            "Governance.Create.Step1.RadioGroup.DFIP.Checked"
+          ).should("be.visible");
+        });
+
+      cy.findByTestId("Governance.Create.Step1.TextArea.NameOfProposal").within(
+        () => {
+          cy.get("textarea").should("have.text", "");
```
Comment (mikhail-zlochevskyi) [inline cypress/e2e/pages/governance/create/index.spec.ts:540]: Can we please fill it up those inputs initially, and then check that they are empty

File: cypress/e2e/pages/governance/create/index.spec.ts [hunk 5]
```diff
@@ -0,0 +1,593 @@
+const viewPorts = ["macbook-16", "ipad-2", "iphone-x"];
+
+viewPorts.forEach((viewPort) => {
+  context(`/governance/create on ${viewPort}`, () => {
+    const nameOfProposal = "Test Proposal";
+    const discussion = "https://github.com/DeFiCh/dfips/issues/243";
```
Comment (mikhail-zlochevskyi) [inline cypress/e2e/pages/governance/create/index.spec.ts:6]: can we also add validation of reddit discussion, that is doesn't throw an error in input, thanks!

File: cypress/e2e/pages/governance/create/index.spec.ts (hunk 4 above)
Comment (veralygit) [reply cypress/e2e/pages/governance/create/index.spec.ts:540]: inputs are filled from previous tests when creating CFP proposal cause this is a CFP -> DFIP switch so was checking if the text fields are empty after the switch

File: cypress/e2e/pages/governance/create/index.spec.ts (hunk 5 above)
Comment (veralygit) [reply cypress/e2e/pages/governance/create/index.spec.ts:6]: added test cases to check for reddit discussion !

File: cypress/e2e/pages/governance/create/index.spec.ts [hunk 6]
```diff
@@ -0,0 +1,593 @@
...
+      cy.findByTestId("Governance.Create.Step1.TextArea.NameOfProposal").should(
+        "be.visible"
+      );
+      cy.findByTestId("Governance.Create.Step1.TextArea.NameOfProposal").within(
+        () => {
+          cy.get("textarea").click().type(nameOfProposal);
+          cy.findByTestId(
+            "Governance.Create.Step1.TextArea.NameOfProposal.ClearForm"
+          ).click();
+          cy.findByTestId("Governance.Create.Step1.TextArea.ErrorMsg").should(
```
Comment (mikhail-zlochevskyi) [inline cypress/e2e/pages/governance/create/index.spec.ts:124]: can we also please add here a check, that Review Proposal button is inactive; before clear the form, thanks!

Comment (veralygit) [reply cypress/e2e/pages/governance/create/index.spec.ts:131]: updated to add checks for review proposal to remain inactive when a single input field is filled

File: cypress/e2e/pages/governance/create/index.spec.ts [hunk 7]
```diff
@@ -0,0 +1,705 @@
...
+        );
+
+        cy.findByTestId("Governance.Create.Step1.Description").should(
+          "have.text",
+          "Enter from GitHub or Reddit the title of the proposal and the type of proposal."
+        );
+      });
+    });
+
+    it("should have step 1 cfp radio group checked and dfip radio unchecked", () => {
```
Comment (kyleleow) [inline cypress/e2e/pages/governance/create/index.spec.ts:71]:
```suggestion
    it("should have step 1 cfp radio group checked and dfip radio unchecked by default", () => {
```

File: cypress/e2e/pages/governance/create/index.spec.ts [hunk 8]
```diff
@@ -0,0 +1,705 @@
...
+          .should("be.visible")
+          .within(() => {
+            cy.findByTestId(
+              "Governance.Create.Step1.RadioGroup.DFIP.Unchecked"
+            ).should("be.visible");
+          });
+      });
+    });
+
+    it("should have step 1 dfip radio group checked and cfp radio group unchecked", () => {
```
Comment (kyleleow) [inline cypress/e2e/pages/governance/create/index.spec.ts:94]:
```suggestion
    it("should be able to check dfip radio group in step 1", () => {
```

File: cypress/e2e/pages/governance/create/index.spec.ts [hunk 9]
```diff
@@ -0,0 +1,705 @@
...
+      });
+      cy.findByTestId(
+        "Governance.Create.Step1.TextArea.ReceivingAddress"
+      ).within(() => {
+        cy.get("textarea").click().type(receivingAddress);
+      });
+      cy.findByTestId("Governance.Create.Step1.ReviewProposal").click();
+    });
+
+    it("should have step 2 header and description", () => {
```
Comment (kyleleow) [inline cypress/e2e/pages/governance/create/index.spec.ts:411]: can we add a test to check that step 1 content is hidden before this test?

File: cypress/e2e/pages/governance/create/index.spec.ts (hunk 7 above)
Comment (veralygit) [reply cypress/e2e/pages/governance/create/index.spec.ts]: updated !

File: cypress/e2e/pages/governance/create/index.spec.ts (hunk 8 above)
Comment (veralygit) [reply cypress/e2e/pages/governance/create/index.spec.ts]: updated !

File: cypress/e2e/pages/governance/create/index.spec.ts (hunk 9 above)
Comment (veralygit) [reply cypress/e2e/pages/governance/create/index.spec.ts:417]: added test to check that steps is hidden

**Review by mikhail-zlochevskyi — DISMISSED** [review]

**Review by kyleleow — DISMISSED** [review]

**Pull request merged**